### PR TITLE
add support for journal upload and remote server

### PR DESCRIPTION
**Pull request by trefzer**
#### Pull Request (PR) description
add support for uploading journal to a remote server

### CODE REVIEW

File: manifests/journal_upload.pp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  if $package_name {
+    stdlib::ensure_packages($package_name)
+  }
+
+  service { 'systemd-journal-upload':
+    ensure => running,
+  }
+  $systemd::journal_upload_settings.each |$option, $value| {
+    ini_setting {
+      $option:
```
Comment (bastelfreak) [inline manifests/journal_upload.pp:22]: I think the following syntax is more common in our code base:
```suggestion
    ini_setting {  $option:
```

Comment (trefzer) [reply manifests/journal_upload.pp:22]: took it from journald.pp, will change.

**Comment by trefzer** [comment]
added journal-remote setting to this merge request to avoid merge conflicts.
This gives the possibility to use combined spec test.

**Comment by kenyon** [comment]
You'll probably need to change the acceptance tests to use Vagrant instead of the default of Docker for systemd to work correctly.

File: types/journaluploadsettings.pp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Matches Systemd journal upload config Struct
```
Comment (bastelfreak) [inline types/journaluploadsettings.pp:1]: to please puppet-strings, please use:
```suggestion
# @summary Matches Systemd journal upload config Struct
```

File: types/journalremotesettings.pp [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Matches Systemd journal remote config Struct
```
Comment (bastelfreak) [inline types/journalremotesettings.pp:1]:
```suggestion
# @summary Matches Systemd journal remote config Struct
```

File: manifests/journal_upload.pp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  Optional[String[1]] $package_name = undef,
+) {
+  assert_private()
+
+  if $package_name {
+    stdlib::ensure_packages($package_name)
+  }
+
+  service { 'systemd-journal-upload':
+    ensure => running,
```
Comment (bastelfreak) [inline manifests/journal_upload.pp:18]: to ensure we don't have a corrective change after a reboot, should we add `enable => true`?

**Comment by trefzer** [comment]
@kenyon probably, but no idea howto do so !

File: manifests/journal_remote.pp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  Optional[String[1]] $package_name = undef,
+) {
+  assert_private()
+
+  if $package_name {
+    stdlib::ensure_packages($package_name)
+  }
+
+  service { 'systemd-journal-remote':
+    ensure => running,
```
Comment (bastelfreak) [inline manifests/journal_remote.pp:18]: should we also set `enable => true,` to put it into autostart? usually we do that.

File: manifests/journal_remote.pp [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+# @param package_name
+#  name of the package to install for the functionality
+#
+class systemd::journal_remote (
+  Optional[String[1]] $package_name = undef,
+) {
+  assert_private()
+
+  if $package_name {
+    stdlib::ensure_packages($package_name)
```
[the remark this answers is not in the document]
Comment (trefzer) [reply manifests/journal_remote.pp:14]: right, but there is currently only one package for upload and remote journal in Debian and RedHat. 
This solution gives you the possibility to einer install upload and remote or both.

File: manifests/journal_remote.pp (hunk 5 above)
Comment (trefzer) [reply manifests/journal_remote.pp:18]: done

File: manifests/journal_remote.pp (hunk 6 above)
[the remark this answers is not in the document]
Comment (bastelfreak) [reply manifests/journal_remote.pp:14]: Oh it's the same package. This packaging is so stupid :(
yes in that case it makes sense to stick to `stdlib::ensure_packages()`

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**